### PR TITLE
fix(onesync): prevent ped spawn in vehicles

### PR DESCRIPTION
**Pull request by dolutattoo**
Sometimes, CreateAutomobile is creating ped inside created vehicles. It only happens when spawning a model which is also spawned by clients in the same area as POPTYPE_RANDOM_PERMANENT (even if this native only create POPTYPE_PERMANENT entities).

### CODE REVIEW

**Comment by VIRUXE** [comment]
Why would you do something like this? That's a personal choice.

File: server/onesync.lua [hunk 1]
```diff
@@ -55,6 +55,18 @@ ESX.OneSync.SpawnVehicle = function(model, coords, heading, cb)
 	CreateThread(function()
 		local entity = Citizen.InvokeNative(`CREATE_AUTOMOBILE`, model, coords.x, coords.y, coords.z, heading)
 		while not DoesEntityExist(entity) do Wait(50) end
+
+		CreateThread(function()
+			local ped = GetPedInVehicleSeat(entity, -1)
+			if ped == 0 then return end
```
Comment (thelindat) [inline server/onesync.lua:61]: Just check if ped > 0 or ~= 0 to perform the loop, rather than having the return.

File: server/onesync.lua [hunk 2]
```diff
@@ -55,6 +55,18 @@ ESX.OneSync.SpawnVehicle = function(model, coords, heading, cb)
...
+			for i=-1, 6 do
```
Comment (thelindat) [inline server/onesync.lua:62]: We can get the number of seats in a vehicle with GetVehicleModelNumberOfSeats

**Review by thelindat — CHANGES_REQUESTED** [review]

File: server/onesync.lua [hunk 3]
```diff
@@ -55,6 +55,18 @@ ESX.OneSync.SpawnVehicle = function(model, coords, heading, cb)
 	CreateThread(function()
 		local entity = Citizen.InvokeNative(`CREATE_AUTOMOBILE`, model, coords.x, coords.y, coords.z, heading)
 		while not DoesEntityExist(entity) do Wait(50) end
+
+		CreateThread(function()
```
[the remark this answers is not in the document]
Comment (dolutattoo) [reply server/onesync.lua]: fixed in next commit

File: server/onesync.lua (hunk 1 above)
Comment (dolutattoo) [reply server/onesync.lua]: done in next commit

File: server/onesync.lua (hunk 2 above)
Comment (dolutattoo) [reply server/onesync.lua:62]: Yes but only client side, right?

**Comment by dolutattoo** [comment]
> Why would you do something like this? That's a personal choice.

We're not overriding the native, you can still use `CreateAutomobile` if you want random ped generation inside your vehicles.
`ESX.OneSync.SpawnVehicle` is the serverside function of `ESX.Game.SpawnVehicle` so it shouldn't spawn anything inside the it.

**Pull request merged**